### ISSUE TITLE
Failed Installation

### ISSUE THREAD

**Issue by astahl7** [hpparvi/PyTransit#25]
Hello,

I recently tried installing PyTransit (for use with K2PS) and have run into some errors. I have asked a friend to try installing PyTransit as well, and he runs into similar issues. I am on Linux Mint and he's on a Mac, and we are both using Anaconda. Here's what I find:

When I try this command from the install instructions:

"python setup.py config_fc --fcompiler=gnu95 --opt="-Ofast" --f90flags="-cpp -fopenmp -static -march=native" build"

I get this error:

/usr/bin/ld: /home/asa/anaconda3/lib/libgomp.a(barrier.o): relocation R_X86_64_32 against `.text' can not be used when making a shared object; recompile with -fPIC
/home/asa/anaconda3/lib/libgomp.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status

It is my understanding that this has to do with interactions between static and dynamic directories. I found a possible solution in modifying ./configure with --enable-shared, but there's no ./configure to do this with. Putting 'sudo' in front of the command gives me "ImportError: No module named numpy.distutils.core", an error which seems to have been solved in the past for scipy and xgboost by using pip install. 

If I ignore the fact that the last command didn't work and try "python setup.py install --user", I get this error:

Warning: src/gimenez.f90:35: Illegal preprocessor directive
Warning: src/gimenez.f90:36: Illegal preprocessor directive
Warning: src/gimenez.f90:37: Illegal preprocessor directive
src/gimenez.f90:271.33:

    real(8), dimension(CHUNK_SIZE, npol) :: d
                                 1
Error: Symbol 'chunk_size' at (1) has no IMPLICIT type

And if I ignore the installation directions and just try using "pip install pytransit", I get this:

#warning "Using deprecated NumPy API, disable it by " \
      ^
...
 Warning: src/gimenez.f90:35: Illegal preprocessor directive
    Warning: src/gimenez.f90:36: Illegal preprocessor directive
    Warning: src/gimenez.f90:37: Illegal preprocessor directive
    src/gimenez.f90:271.33:
    
        real(8), dimension(CHUNK_SIZE, npol) :: d
                                     1
    Error: Symbol 'chunk_size' at (1) has no IMPLICIT type

Both of these install methods give this as well:

/home/asa/anaconda3/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'extra_options'

and the config_fc command gives

/home/asa/anaconda3/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)

as well.

As for my friend, he gets the "deprecated NumPy API" warning from the pip install and from the config_fc command. The pip install fails to complete, and the setup.py install --user gives him the same 'extra options' and 'install requires' warnings. It finishes installing, but he still can't import the module in python. 

That about sums it up. I want to thank you for making these very helpful codes, and for taking the time to look into this. I am relatively new to linux and bug reporting in general, so if there is any more information you need for me or anything I can do to help, please don't hesitate to ask.

**Comment by dbrown-astro** [comment]
I am also getting the same problem when I try to install the module currently.

**Comment by astahl7** [comment]
I fixed this issue, but unfortunately I cannot provide a specific set of instructions for what solved my problem because I basically just shotgunned a bunch of solutions at the problem until it worked. 

However, I can say that my problem centered around miscommunications and misallocations between the python Anaconda was using, the built-in python on my unix system, pip, pip3, an old version of gfortran on my system, an updated version, and the use of admin privileges. A mess, I know. I uninstalled Anaconda and focused on my built-in Python 2, making sure that it was the default and was being correctly forwarded to. Then I made sure my preferences favored the use of the updated gfortran and that it was correctly communicating with python 2. Then I was able to use sudo pip commands to install pytransit.

Perhaps some of that was unnecessary or unrelated to the problem I had. But some combination of all that stuff fixed my issue (though I do have to run python with admin privileges in order for pytransit to import...). If you have any specific questions, dbrown-astro, I can try to answer them - but I'm closing this thread.
